main.tf

```diff
@@ -0,0 +1,138 @@
+/**
+ * Creates a Lambda function with associated role and policies that
+ * will run the packer-janitor tool to get rid of orphaned Packer
+ * instances and their associated resources.
+ *
+ * Creates the following resources:
+ *
+ * * Lambda function
+ * * IAM role and policies to describe and delete instances, keypairs,
+ *   and security groups, as well as write messages to Cloudwatch Logs
+ * * Cloudwatch Logs group
+ * * Cloudwatch Event to regularly run cleanup job
+ *
+ * ## Usage
+ *
+ * ```hcl
+ * module "packer-janitor-lambda" {
+ *   source                 = "trussworks/lambda-packerjanitor/aws"
+ *   packer_resource_delete = "true"
+ *   packer_timelimit       = "4"
+ *   job_identifier         = "sample_app"
+ *   s3_bucket              = "sample_app_lambdas"
+ *   version_to_deploy      = "2.8"
+ * }
+ * ```
+ *
+ * For more details on the capabilities of the packer-janitor tool, as
+ * well as how to deploy it, see <https://github.com/trussworks/truss-aws-tools>.
+ */
+
+locals {
+  pkg  = "truss-aws-tools"
+  name = "packer-janitor"
+}
+
+data "aws_region" "current" {}
+
+# This is the main policy this job is going to need.
+data "aws_iam_policy_document" "main" {
+  # Allow describing instances, keypairs, and security groups.
+  statement {
+    sid    = "Describes"
+    effect = "Allow"
+
+    actions = [
+      "ec2:DescribeInstances",
+      "ec2:DescribeKeyPairs",
+      "ec2:DescribeSecurityGroups",
+    ]
+
+    resources = ["*"]
+  }
+
+  # Allow terminating EC2 Instances
+  statement {
+    sid    = "TerminateInstances"
+    effect = "Allow"
+
+    actions = [
+      "ec2:TerminateInstances",
+    ]
+
+    resources = ["*"]
```

Comment:
Do the instances get tagged such that it's clear they are from Packer? Like, maybe the "Name" tag gets set to "Packer"? If so, we should add a conditional so the lambda can only delete those.

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-supported-iam-actions-resources.html

Reply:
Yep, good catch. They all have the name "Packer Builder".

Reply:
Added a conditional to this in the TerminateInstances permission now.